### DISCUSSION TITLE
Version of SKP firmware that implements SidBlaster protocol

### DISCUSSION THREAD

**Discussion by magwed** [frntc/SIDKick-pico#13] [Ideas]
After looking into devices that use SID it seem to be a standard called SIDBlaster which implements a hardware device which can receive real-time messages over serial line/network port. Would it be feasible to have a SidBlaster firmware for SKP? Normally you have another device (f.e. a PIC-device) that implements the USB-part and then use a real SID-chip. So SKP could be inserted into such a card at it would work. But why even implement C64-bus if there aren't any C64? Instead maybe SKP could implement everything, both SID-emulation and serial port interpreter and show up like a SID-device over USB-serial.

Either it could listen to both C64-bus and RX-serial line or switchable at build-time to either implement serial line listener or C64-bus listener.

**Comment by frntc** [comment]
I've no experience with the SIDBlaster, but as far as I understand it's meant to use a real SID with a PC. So if you use emulation (as SKpico obviously does), why not let the emulation run on the PC? ;-)

**Reply by magwed** [reply]
That is absolutely correct, I also thought about that! :-) But still, I think this would be a nice extra. 
Also it makes it possible to "non"-electronics people to use/test your great work, without having to be a tech-wizard, thus without having one and having to open your old computer. I think it would be cool!

**Reply by frntc** [reply]
Sure it's a nice extra, but I have to use the scarce resource "time" carefully :-) There are a few things emulation-wise that I would like to improve and also continue working on a smaller version (already presented on forum64.de) first... then we'll see!
